src/reader/multilinestring.rs

```diff
@@ -92,16 +92,16 @@ impl<'a> MultiLineStringTrait for MultiLineString<'a> {
     }
 
     unsafe fn line_string_unchecked(&self, i: usize) -> Self::LineStringType<'_> {
-        *self.wkb_line_strings.get_unchecked(i)
+        self.wkb_line_strings.get_unchecked(i)
     }
 }
 
-impl<'a> MultiLineStringTrait for &MultiLineString<'a> {
+impl<'a, 'b> MultiLineStringTrait for &'b MultiLineString<'a> {
     type T = f64;
-    type LineStringType<'b>
-        = LineString<'a>
+    type LineStringType<'c>
+        = &'b LineString<'a>
     where
-        Self: 'b;
+        Self: 'c;
```

Comment:
I trust that this is correct but once we get to three lifetimes I struggle to reason about it 😅